test/ui/OmniboxResultListTest.ts

```diff
@@ -0,0 +1,166 @@
+import * as Mock from '../MockEnvironment';
+import { OmniboxResultList, IOmniboxResultListOptions } from '../../src/ui/OmniboxResultList/OmniboxResultList';
+import { FakeResults } from '../Fake';
+import { $$ } from '../../src/utils/Dom';
+import { IQueryResults } from '../../src/rest/QueryResults';
+import { Simulate } from '../Simulate';
+import { Template } from '../../src/ui/Templates/Template';
+import { get } from '../../src/ui/Base/RegisteredNamedMethods';
+import { Quickview } from '../../src/ui/Quickview/Quickview';
+import { ResultLink } from '../../src/ui/ResultLink/ResultLink';
+
+export function OmniboxResultListTest() {
+  describe('OmniboxResultListTest', () => {
+    let test: Mock.IBasicComponentSetup<OmniboxResultList>;
+    let results: IQueryResults;
+
+    beforeEach(() => {
+      test = Mock.basicComponentSetup<OmniboxResultList>(OmniboxResultList);
+      results = FakeResults.createFakeResults();
+    });
+
+    it('should support building results', async done => {
+      const built = await test.cmp.buildResults(results);
+      expect(built.length).toEqual(results.results.length);
+      done();
+    });
+
+    it('should make all results selectable by the omnibox', async done => {
+      const built = await test.cmp.buildResults(results);
+      built.forEach(element => {
+        expect($$(element).hasClass('coveo-omnibox-selectable')).toBe(true);
+      });
+      done();
+    });
+
+    it('should set the no-result-suggestion propery on each result for the magicbox', async done => {
+      const built = await test.cmp.buildResults(results);
+      built.forEach(element => {
+        expect(element['no-text-suggestion']).toBeTruthy();
+      });
+      done();
+    });
+
+    describe('when selecting an element', () => {
+      let spyOnSelect: jasmine.Spy;
+
+      beforeEach(() => {
+        spyOnSelect = jasmine.createSpy('onClick');
+        test.cmp.options.onSelect = spyOnSelect;
+        Simulate.omnibox(test.env);
+      });
+
+      it('should call the onSelect option when clicking on each element of the result list', async done => {
+        const built = await test.cmp.buildResults(results);
+
+        built.forEach(element => {
+          $$(element).trigger('click');
+        });
+        expect(spyOnSelect).toHaveBeenCalledTimes(10);
```

Comment:
We could use the length of the results list length here, it would avoid to update the test in case the length of that result list changes.